tests/v2/validation/charts/istio_test.go

```diff
@@ -0,0 +1,303 @@
+	require.NoError(i.T(), err)
+
+	i.T().Log("Validating kiali and jaeger endpoints are accessible")
+	kialiResult, err := charts.GetChartCaseEndpoint(client, client.RancherConfig.Host, kialiPath, true)
+	require.NoError(i.T(), err)
+	require.True(i.T(), kialiResult.Ok)
+
+	tracingResult, err := charts.GetChartCaseEndpoint(client, client.RancherConfig.Host, tracingPath, true)
+	require.NoError(i.T(), err)
+	require.True(i.T(), tracingResult.Ok)
+
+	//Get a random worker node' public external IP of a specific cluster
+	nodeCollection, err := client.Management.Node.List(&types.ListOpts{Filters: map[string]interface{}{
+		"clusterId": i.project.ClusterID,
+	}})
+	require.NoError(i.T(), err)
+	workerNodePublicIPs := []string{}
+	for _, node := range nodeCollection.Data {
+		workerNodePublicIPs = append(workerNodePublicIPs, node.Annotations["rke.cattle.io/external-ip"])
+	}
+	randWorkerNodePublicIP := workerNodePublicIPs[rand.Intn(len(workerNodePublicIPs))]
+	istioGatewayHost := randWorkerNodePublicIP + ":" + exampleAppPort
+
+	i.T().Log("Validating example app is accessible")
+	exampleAppResult, err := charts.GetChartCaseEndpoint(client, istioGatewayHost, exampleAppProductPagePath, false)
+	require.NoError(i.T(), err)
+	require.True(i.T(), exampleAppResult.Ok)
+
+	i.T().Log("Validating example app has three different reviews bodies")
+	doesContainFirstPart, err := getChartCaseEndpointUntilBodyHas(client, istioGatewayHost, exampleAppProductPagePath, firstReviewBodyPart)
+	require.NoError(i.T(), err)
+	require.True(i.T(), doesContainFirstPart)
+
+	doesContainSecondPart, err := getChartCaseEndpointUntilBodyHas(client, istioGatewayHost, exampleAppProductPagePath, secondReviewBodyPart)
+	require.NoError(i.T(), err)
+	require.True(i.T(), doesContainSecondPart)
+
+	doesContainThirdPart, err := getChartCaseEndpointUntilBodyHas(client, istioGatewayHost, exampleAppProductPagePath, thirdReviewBodyPart)
+	require.NoError(i.T(), err)
+	require.True(i.T(), doesContainThirdPart)
+
+	i.T().Log("Cleaning up the case resources")
+}
+
+func (i *IstioTestSuite) TestUpgradeIstioChart() {
+	subSession := i.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := i.client.WithSession(subSession)
+	require.NoError(i.T(), err)
+
+	catalogClient, err := client.GetClusterCatalogClient(i.project.ClusterID)
+	require.NoError(i.T(), err)
+
+	//Change istio install option version to previous version of the latest version
+	versionsList, err := client.Catalog.GetListChartVersions(charts.RancherIstioName)
+	require.NoError(i.T(), err)
+	require.Greaterf(i.T(), len(versionsList), 2, "There should be at least 2 versions of the chart")
+	versionBeforeLatest := versionsList[1]
+	i.chartInstallOptions.istio.Version = versionBeforeLatest
+
+	i.T().Log("Installing istio chart with the last but one version")
+	err = charts.InstallRancherIstioChart(client, i.chartInstallOptions.istio, i.chartFeatureOptions.istio)
```

Comment:
I want to understand, in this upgrade test you wouldn't need to install monitoring as well? I'm little confused

Reply:
As far as I know, In the P0 test case; We need the rancher monitoring chart to be installed to validate Kiali and Jaeger. In the upgrade test case, it simply checks if the Istio chart with the older version is upgraded successfully to the latest version.